test/support/factory.ex

```diff
@@ -150,15 +150,31 @@ defmodule Wanda.Factory do
     execution_id = Map.get(attrs, :execution_id, Faker.UUID.v4())
     group_id = Map.get(attrs, :group_id, Faker.UUID.v4())
 
+    payload =
+      Map.get(attrs, :payload, build(:result, execution_id: execution_id, group_id: group_id))
+
     %ExecutionResult{
       execution_id: execution_id,
       group_id: group_id,
-      payload: build(:result, execution_id: execution_id, group_id: group_id),
+      status: Map.get(attrs, :status, :running),
+      targets: Map.get(attrs, :targets, []),
+      payload: payload,
       inserted_at: Map.get(attrs, :inserted_at, DateTime.utc_now()),
-      updated_at: Map.get(attrs, :updated_at, DateTime.utc_now())
+      updated_at: Map.get(attrs, :updated_at, DateTime.utc_now()),
+      completed_at: Map.get(attrs, :completed_at, nil)
     }
   end
 
+  def running_execution_result_factory(_attrs) do
```

Comment:
ExMachina provides better ways to do this.
Check this: https://github.com/thoughtbot/ex_machina#flexible-factories-with-pipes

We could have `with_running_state` function

Reply:
At the end these 2 factories are not needed, right?

Reply:
No indeed, removed.